### PR TITLE
[@types/k6] k6 release v0.50.0

### PR DESCRIPTION
**Pull request by oleiade**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/k6/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "private": true,
     "name": "@types/k6",
-    "version": "0.49.9999",
+    "version": "0.50.0",
```
Comment (codershiba) [inline types/k6/package.json:4]:
```suggestion
    "version": "0.50.9999",
```

File: types/k6/experimental/browser.d.ts [hunk 2]
```diff
@@ -2914,6 +3011,34 @@ export interface Page {
...
+     * @param selector A selector to search for an element. If there are multiple
+     * elements satisfying the selector, the first will be used.
+     * @param files
+     * @param options
+     */
+    setInputFiles(selector: string, files: File | File[], options?: {
+        /**
+         * Maximum time in milliseconds. Pass 0 to disable the timeout. Default
+         * is overridden by the setDefaultTimeout option on {@link BrowserContext} or
+         * {@link Page}. Defaults to 30000.
```
Comment (codershiba) [inline types/k6/experimental/browser.d.ts:3031]:
```suggestion
         * {@link Page}
         * @default 30000
```

File: types/k6/experimental/browser.d.ts [hunk 3]
```diff
@@ -2914,6 +3011,34 @@ export interface Page {
...
+         */
+        timeout?: number;
+
+        /**
+         * If set to `true` and a navigation occurs from performing this action, it
+         * will not wait for it to complete. Defaults to `false`.
```
Comment (codershiba) [inline types/k6/experimental/browser.d.ts:3037]:
```suggestion
         * will not wait for it to complete.
         * @default false
```

File: types/k6/experimental/browser.d.ts [hunk 4]
```diff
@@ -1479,6 +1548,34 @@ export interface Frame {
...
```
Comment (codershiba) [inline types/k6/experimental/browser.d.ts:1574]:
```suggestion
         * will not wait for it to complete.
         * @default false
```

File: types/k6/experimental/browser.d.ts [hunk 5]
```diff
@@ -1479,6 +1548,34 @@ export interface Frame {
...
+     * @param selector A selector to search for an element. If there are multiple
+     * elements satisfying the selector, the first will be used.
+     * @param files
+     * @param options
+     */
+    setInputFiles(selector: string, files: File | File[], options?: {
+        /**
+         * Maximum time in milliseconds. Pass 0 to disable the timeout. Default
+         * is overridden by the setDefaultTimeout option on {@link BrowserContext} or
+         * {@link Page}. Defaults to 30000.
```
Comment (codershiba) [inline types/k6/experimental/browser.d.ts:1568]:
```suggestion
         * {@link Page}
         * @default 30000
```

File: types/k6/experimental/browser.d.ts [hunk 6]
```diff
@@ -1131,6 +1175,31 @@ export interface ElementHandle extends JSHandle {
...
+         */
+        timeout?: number;
+
+        /**
+         * If set to `true` and a navigation occurs from performing this action, it
+         * does not wait for it to complete. Defaults to `false`.
```
Comment (codershiba) [inline types/k6/experimental/browser.d.ts:1198]:
```suggestion
         * does not wait for it to complete.
         * @default false
```

File: types/k6/experimental/browser.d.ts [hunk 7]
```diff
@@ -1131,6 +1175,31 @@ export interface ElementHandle extends JSHandle {
...
+     *
+     * The {@link ElementHandle | element handle} must be an [input element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input).
+     * @param files
+     * @param options
+     */
+    setInputFiles(files: File | File[], options?: {
+        /**
+         * Maximum time in milliseconds. Pass 0 to disable the timeout. Default
+         * is overridden by the setDefaultTimeout option on {@link BrowserContext} or
+         * {@link Page}. Defaults to 30000.
```
Comment (codershiba) [inline types/k6/experimental/browser.d.ts:1192]:
```suggestion
         * {@link Page}.
         * @default 30000
```

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/k6/test/webcrypto.ts
1. types/k6/test/browser.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0b69ea318b87522fcd1cf1d64cbc2fc0eb986d9a
</p>

**Review by ghost — APPROVED** [review]

**Review by mstoykov — APPROVED** [review]

**Pull request merged**